CorsixTH/Lua/ui.lua

```diff
@@ -1066,6 +1066,14 @@ function UI:removeWindow(closing_window)
   end
 end
 
+--! Function to check if we have any must pause windows open
+--!return (bool) Returns true if a must pause window is found
+function UI:checkForMustPauses()
+  for i in pairs(self.windows) do
+    if self.windows[i].mustPause() then return true end
+  end
+end
```

Comment:
`nil` is not a valid boolean value to return.

The function name doesn't cover the computation. "check for" reads to me as "check if everything is ok" (and return true/false). Try to be more precise in what you compute.

Boolean functions often have "isX" or "hasX" as name, eg "haveMustPauseWindows()" is very concrete and also explains what a "true" as returned value means.

A sign here is when your longer explanation mentions relevant things not expressed in the function name.